client/src/Pos/Client/Txp/Util.hs

```diff
@@ -557,6 +590,32 @@ createTx pendingTx utxo ss outputs addrData =
     createGenericTxSingle pendingTx (\i o -> Right $ makePubKeyTx ss i o)
     OptimizeForSecurity utxo outputs addrData
 
+-- | Create generic unsigned Tx
+createGenericUnsignedTx
+    :: TxCreateMode m
+    => PendingAddresses
+    -> (TxOwnedInputs TxOut -> TxOutputs -> Either TxError Tx)
+    -> InputSelectionPolicy
+    -> Utxo
+    -> TxOutputs
+    -> m (Either TxError (Tx,NonEmpty TxOut))
+createGenericUnsignedTx pendingTx creator inputSelectionPolicy utxo outputs =
+    runTxCreator inputSelectionPolicy $ do
+        (inps, outs) <- prepareInpsOutsForUnsignedTx pendingTx utxo outputs
+        tx <- either throwError return $ creator inps outs
+        pure (tx, map fst inps)
+
+-- | Create unsigned Tx
+createUnsignedTx
+    :: TxCreateMode m
+    => PendingAddresses
+    -> InputSelectionPolicy
+    -> Utxo
+    -> TxOutputs
+    -> m (Either TxError (Tx,NonEmpty TxOut))
+createUnsignedTx pendingTx groupInputs utxo outputs =
+    createGenericUnsignedTx pendingTx makeUnsignedAbstractTx groupInputs utxo outputs
+
```

Comment:
There're a lot of same Tx-related functions, like `createTx`, `prepareTx` or `makeTx`. :-) Kostas made it for simplicity, by analogy with existent code.